### PR TITLE
wip: RHMAP-18943 macos-prereqs review

### PR DESCRIPTION
**Pull request by pmacko1**
splinter from https://github.com/aerogear/aerogear.org/pull/683/files
https://issues.jboss.org/browse/RHMAP-18943

# I will close this, remove unwanted merge commits in # and raise a jira to follow upon the convo re macos sudo password

### CODE REVIEW

File: docs/digger/installation/macos-prereqs.asciidoc [hunk 1]
```diff
@@ -1,35 +1,30 @@
-[[macos-server]]
-=== macOS Server pre-requisites
+[[macos-prereqs]]
+==== macOS prerequisites (optional)
 
-NOTE: This section is required for building iOS applications. Skip to link:#run-digger-installer[Run Digger Installer] if you are NOT building for iOS or DO NOT wish to configure a mac machine for you digger installation.
```
Comment (laurafitzgerald) [inline docs/digger/installation/macos-prereqs.asciidoc]: Why is this note being removed?

File: docs/digger/installation/macos-prereqs.asciidoc [hunk 2]
```diff
@@ -1,35 +1,30 @@
...
+* 2 CPU Cores
+* 30GB minimum disk space and and additional 20GB per installed XCode version
 
 
-The sections below outline the required state for the macOS node. Ansible requires SSH access to the macOS server in order to correctly provision the machine to support iOS builds. To enable this, it is required that
-a user with sudo permissions and SSH access exists on the target machine. Read the following sections and ensure these conditions are met to enable correct configuration of your macOS server via the Ansible installation.
+===== Create SSH User for Ansible
 
-====== Create SSH User
+Ansible requires an SSH access to the macOS server in order to correctly provision the server to support iOS builds. Create a user with sudo permissions and SSH access on the macOS server.
```
Comment (laurafitzgerald) [inline docs/digger/installation/macos-prereqs.asciidoc]: remove `an` here

File: docs/digger/installation/macos-prereqs.asciidoc (hunk 1 above)
Comment (pmacko1) [reply docs/digger/installation/macos-prereqs.asciidoc]: i think the note (optional) contains the info in this paragraph @laurafitzgerald

Comment (laurafitzgerald) [reply docs/digger/installation/macos-prereqs.asciidoc]: @pmacko1 the motivation of adding this is to make it clear to the user where they should go if they wish to only build Android applications. If this is being removed can we keep a link somewhere in the text to show the user. The page itself is quite big once it's built so would be difficult for the user to pick this out themselves.

Comment (pmacko1) [reply docs/digger/installation/macos-prereqs.asciidoc]: @laurafitzgerald a very good point. This is kinda overlooked when focusing on technical content. Will there be a "sections" bar in the final render? (e.g. https://docs.openshift.com/container-platform/3.7/welcome/index.html left column). I can add a TOC in the beginning of the installation doc if that will help

Comment (laurafitzgerald) [reply docs/digger/installation/macos-prereqs.asciidoc]: @pmacko1 yes there is sections on the aerogear website when it is build. You can see this here: https://aerogear.org/docs/digger/installation/ Let's not add a TOC as it doesn't fit with the rest of the site.

Comment (pmacko1) [reply docs/digger/installation/macos-prereqs.asciidoc]: note readded and polished
resolved

File: docs/digger/installation/macos-prereqs.asciidoc (hunk 2 above)
Comment (pmacko1) [reply docs/digger/installation/macos-prereqs.asciidoc]: resolved

**Comment by pmacko1** [comment]
given this another review.
"
Set `ansible_sudo_pass` variable or `--ask-sudo-pass` flag with the user password to enable the installer to work with root permissions.
" with which command should this flag be passed @laurafitzgerald ?

**Comment by laurafitzgerald** [comment]
@pmacko1 not sure where the above comment refers to. Can you add a link to it?

**Comment by pmacko1** [comment]
@laurafitzgerald https://github.com/aerogear/aerogear.org/pull/704/files#diff-7b552d92b91145930a93588b4c11bb2fR133

**Comment by laurafitzgerald** [comment]
@pmacko1 That flag should be passed when running the installer with the macOS section. It should be included at the step where the user is running the command at the start. here: https://github.com/aerogear/aerogear.org/pull/711/files#diff-489bedbc7d27bc1215285e65ca88ce48R18

**Comment by pmacko1** [comment]
@laurafitzgerald what is the reason to implement 2 routes for providing a password? What is the preferred one-> how should we document this?

**Comment by laurafitzgerald** [comment]
Using the variable reduces the amount of interaction that the user has to have with the playbook when it runs i.e. (not having to enter the password during the playbook). however it does mean that they have to put a password into a file. If we recommend this then we should also mention the ansible vault option.
otherwise we can use the `--ask-vault-pass`. @finp from a docs perspective is it better to give the user one direct path or give them the option of two depending on which they prefer to use? both are possible with ansible?

**Comment by pmacko1** [comment]
ping @finp @secondsun @matskiv

**Comment by pmacko1** [comment]
@laurafitzgerald could you elab on your last comment pls?

**Comment by pmacko1** [comment]
I will merge this and raise a jira to follow upon the convo re macos sudo password

**Pull request closed without merging**